### PR TITLE
Reader: add reader/settings to state

### PR DESCRIPTION
**Pull request by bluefuton**
Following on from https://github.com/Automattic/wp-calypso/pull/11282#issuecomment-293311636, we need a way of storing flags against a user to show which messages they've seen in the following intro component.

Initially, we'll be using `is_new_reader` to show the basic introduction to new users.

### The masterplan

These flags may not be Calypso-specific, so I was wary of adding them to `calypso_preferences`. Instead, we piggyback on existing calls to `/me/settings` by receiving `PREFERENCES_FETCH_SUCCESS`, which has the whole API response.

We grab any keys under `reader` and store them in state.reader.settings.items.

We'll then add a new action called deleteReaderSetting, which remove these flags when the user dismisses the message (and removes the flag from state on success).

### CODE REVIEW

File: client/state/reader/settings/reducer.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import { READER_USER_SETTING_KEY } from './constants';
+
+/**
+ * Returns the current user's Reader settings. These come from the reader key on /me/settings.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function items( state = {}, action ) {
```
Comment (samouri) [inline client/state/reader/settings/reducer.js]: default to null?

**Comment by mtias** [comment]
I think it should be ok to use the existing preferences. (We use it for recent sites which is now shared with mobile apps, afaik).

**Comment by bluefuton** [comment]
Thanks @mtias!

My other motivation to keep them separate was that `is_new_reader` needs to be set when the user is created (in the `/users/new` endpoint), e.g.

```update_user_attribute( $user->ID, 'calypso_preferences', [ 'reader' => [ 'is_new_reader' ] => true ] ] );```

I believe that'd wipe out anything in calypso_preferences - that _should_ be okay at the point of creating a brand new user but wanted to double check first (@aduth?)

**Comment by aduth** [comment]
I'm not aware of anything else that would initialize `calypso_preferences` that we risk overriding with your logic. Perhaps for future maintainability's sake we ought to consolidate it as part of a generalized user initialization. I've not yet looked at what other attributes we initialize.

**Comment by bluefuton** [comment]
Thanks all. I'm going to close this out and use Calypso preferences for the `is_new_reader` flag 👍

**Pull request closed without merging**